webrender/src/renderer.rs

```diff
@@ -1833,7 +1833,10 @@ impl Renderer {
 
     /// Returns the Epoch of the current frame in a pipeline.
     pub fn current_epoch(&self, pipeline_id: PipelineId) -> Option<Epoch> {
-        self.pipeline_info.epochs.get(&pipeline_id).cloned()
+        self.pipeline_info.epochs
```

Comment:
is the current epoch useful for non-built pipelines?

Reply:
Yes, knowing the epoch lets us know which gecko transaction has been completed, and lets us fire the MozAfterPaint event on the gecko side, which is the desired goal of this change.

Reply:
but if the pipeline isn't built, it's not painted either, so how does `MozAfterPaint` make sense?

Reply:
Because Gecko. You could consider it a bug, but in this case it's easier to bug-compatible with gecko than rewrite large chunks of the gecko frontend :(

Reply:
Yeah it should really be called `MozAfterPrepareToComposite` or `MozReadyToCompositeInstantly`.

Reply:
ok, @mstange , but it's not even prepared/ready if it's not built!

Reply:
Maybe MozCompositorHasSomeIdeaThatThisExists :)

Reply:
Actually, sorry, I was mixing up `MozAfterPaint` with `MozLayerTreeReady`. I concur with kats now; Gecko is being stupid and using `MozAfterPaint` in a confused way, but matching the bug is the easy way out.